pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1546,9 +1547,21 @@ func (vs *vSphereVMProvider) vmCreateDoPlacement(
 				vmopv1.VirtualMachineConditionPlacementReady,
 				"NotReady",
 				retErr)
+		} else {
+			pkgcnd.MarkTrue(
+				vmCtx.VM,
+				vmopv1.VirtualMachineConditionPlacementReady)
 		}
 	}()
 
+	if pkgcfg.FromContext(vmCtx).Features.VMGroups &&
+		vmCtx.VM.Spec.GroupName != "" {
```

Comment:
If we have a validation webhook that prevents someone from setting `spec.groupName` unless the feature is enabled then we don't need the feature check here; checking whether or not `spec.groupName` is empty is sufficient.